crates/core/src/network/networking.rs

```diff
@@ -105,7 +112,7 @@ impl NetworkSender for Networking {
                         to,
                         e,
                         attempts + 1,
-                        DEFAULT_MAX_CONNECT_ATTEMPTS
+                        self.max_retry_attempts
```

Comment:
What if `max_retry_attempts == 0`? Might look a bit weird with `next retry is attempt 2/0`.

Reply:
I'll fix in the final revision; agree this is unsightly!

Reply:
I remember now that the reason why I decided this is fine is both retry-capable policies have a `NonZeroUsize` max attempts; if we have a no-retry policy then we should never print this message out. But I wish this was more explicit in the code.